src/opower/utilities/coautilities.py

```diff
@@ -0,0 +1,137 @@
+"""City of Austin Utilities."""
+import re
+from typing import Optional
+
+import aiohttp
+from yarl import URL
+
+from ..const import USER_AGENT
+from ..exceptions import InvalidAuth
+from .base import UtilityBase
+from .helpers import get_form_action_url_and_hidden_inputs
+
+
+class COAUtilities(UtilityBase):
+    """City of Austin Utilities."""
+
+    @staticmethod
+    def name() -> str:
+        """Distinct recognizable name of the utility."""
+        return "City of Austin Utilities"
+
+    @staticmethod
+    def subdomain() -> str:
+        """Return the opower.com subdomain for this utility."""
+        return "coa"
+
+    @staticmethod
+    def timezone() -> str:
+        """Return the timezone.
+
+        Should match the siteTimeZoneId of the API responses.
+        """
+        return "America/Chicago"
+
+    @staticmethod
+    def is_dss() -> bool:
+        """Check if Utility using DSS version of the portal."""
+        return True
+
+    @staticmethod
+    async def async_login(
+        session: aiohttp.ClientSession,
+        username: str,
+        password: str,
+        optional_mfa_secret: Optional[str],
+    ) -> Optional[str]:
+        """Login to the utility website."""
+        # Get cookies
+        await session.get(
+            "https://coautilities.com/wps/wcm/connect/occ/coa/home",
+            headers={"User-Agent": USER_AGENT},
+        )
+
+        # Auth using username and password on coautilities
+        url = (
+            "https://coautilities.com/pkmslogin.form?/isam/sps/OPowerIDP_DSS/saml20/logininitial?"
+            "RequestBinding=HTTPPost&"
+            "NameIdFormat=email&"
+            "PartnerId=opower-coa-dss-webUser&"
+            "Target=https://dss-coa.opower.com"
+        )
+
+        async with session.post(
+            url,
+            headers={"User-Agent": USER_AGENT},
+            data={
+                "username": username,
+                "password": password,
+                "login-form-type": "pwd",
+            },
+        ) as response:
+            await response.text()
+            if "PD-S-SESSION-ID-PCOAUT" not in session.cookie_jar.filter_cookies(
+                URL("https://coautilities.com")
```

Comment:
If I'll pass just a string `mypy --install-types .` will return an error:
```
error: Argument 1 to "filter_cookies" of "AbstractCookieJar" has incompatible type "str"; expected "URL"  [arg-type]
```
This is because of signature of the function at aiohttp:
```
@abstractmethod
def filter_cookies(self, request_url: URL) -> "BaseCookie[str]":
    """Return the jar's cookies filtered by their attributes."""
```